sklearn/model_selection/tests/test_validation.py

```diff
@@ -1194,7 +1196,7 @@ def test_validation_curve():
             MockEstimatorWithParameter(), X, y, param_name="param",
             param_range=param_range, cv=2
         )
-    if len(w) > 0:
+    if len(w) > 1:
```

Comment:
I think this is too loose. You could test != 1, with a comment on why you should expect 1 warning, or, better, you can remove the expected warning from w